chained/https_impl.go

```diff
@@ -82,6 +85,8 @@ func (impl *httpsImpl) dialServer(op *ops.Op, ctx context.Context) (net.Conn, er
 			if impl.tlsClientHelloSplitting {
 				tcpConn = hellosplitter.Wrap(tcpConn, splitClientHello)
 			}
+			tcpConn = tlsFragConn(tcpConn, impl.proxyConfig)
```

Comment:
This makes it look like we always fragment.  I think it'd be more readable if the setting was checked explicitly here, like the other settings.  Not a huge deal though.

Reply:
Yeah it's almost more like a filter, but tbh I don't love the duplicate logic of the other approach